src/callbacks/analysis.jl

```diff
@@ -220,109 +220,110 @@ function (analysis_callback::AnalysisCallback)(integrator)
             (:l2_error, :linf_error, :conservation_error, :residual))
-        print(" Variable:    ")
+        mpi_isroot() && print(" Variable:    ")
         for v in eachvariable(equations)
-          @printf("   %-14s", varnames_cons(equations)[v])
+          mpi_isroot() && @printf("   %-14s", varnames_cons(equations)[v])
         end
-        println()
+        mpi_println()
       end
 
       # Calculate L2/Linf errors, which are also returned by analyze_solution
       l2_error, linf_error = calc_error_norms(u, t, analyzer, semi)
 
       # L2 error
       if :l2_error in analysis_errors
-        print(" L2 error:    ")
+        mpi_isroot() && print(" L2 error:    ")
         for v in eachvariable(equations)
-          @printf("  % 10.8e", l2_error[v])
-          analysis_callback.save_analysis && @printf(io, "  % 10.8e", l2_error[v])
+          mpi_isroot() && @printf("  % 10.8e", l2_error[v])
+          analysis_callback.save_analysis && mpi_isroot() && @printf(io, "  % 10.8e", l2_error[v])
         end
-        println()
+        mpi_println()
       end
 
       # Linf error
       if :linf_error in analysis_errors
-        print(" Linf error:  ")
+        mpi_isroot() && print(" Linf error:  ")
         for v in eachvariable(equations)
-          @printf("  % 10.8e", linf_error[v])
-          analysis_callback.save_analysis && @printf(io, "  % 10.8e", linf_error[v])
+          mpi_isroot() && @printf("  % 10.8e", linf_error[v])
+          analysis_callback.save_analysis && mpi_isroot() && @printf(io, "  % 10.8e", linf_error[v])
         end
-        println()
+        mpi_println()
       end
 
       # Conservation errror
       if :conservation_error in analysis_errors
         @unpack initial_state_integrals = analysis_callback
         state_integrals = integrate(integrator.u, semi)
 
-        print(" |∑U - ∑U₀|:  ")
+        mpi_isroot() && print(" |∑U - ∑U₀|:  ")
         for v in eachvariable(equations)
           err = abs(state_integrals[v] - initial_state_integrals[v])
-          @printf("  % 10.8e", err)
-          analysis_callback.save_analysis && @printf(io, "  % 10.8e", err)
+          mpi_isroot() && @printf("  % 10.8e", err)
+          analysis_callback.save_analysis && mpi_isroot() && @printf(io, "  % 10.8e", err)
         end
-        println()
+        mpi_println()
       end
 
       # Residual (defined here as the vector maximum of the absolute values of the time derivatives)
       if :residual in analysis_errors
-        print(" max(|Uₜ|):   ")
+        mpi_isroot() && print(" max(|Uₜ|):   ")
         for v in eachvariable(equations)
           # Calculate maximum absolute value of Uₜ
-          @views res = maximum(abs, view(du, v, ..))
-          @printf("  % 10.8e", res)
-          analysis_callback.save_analysis && @printf(io, "  % 10.8e", res)
+          res = maximum(abs, view(du, v, ..))
+          res = MPI.Reduce!(Ref(res), max, mpi_root(), mpi_comm())[]
```

Comment:
An MPI routine should IMHO only be called from either a pure-MPI function or hidden behind `mpi_isparallel()`:
```suggestion
          if mpi_isparallel()
            res = MPI.Reduce!(Ref(res), max, mpi_root(), mpi_comm())[]
          end
```